gpkit/model.py

```diff
@@ -544,3 +546,53 @@ def form_program(programType, signomials, verbosity=2):
         return sp, sp.localsolve
     else:
         raise ValueError("unknown program type %s." % programType)
+
+
+def simplify_and_mmap(signomials, subs):
+    """Simplifies a list of signomials and returns them with their mmaps.
+
+    Arguments
+    ---------
+    signomials : list of Signomials
+
+    subs : dict
+        Substitutions to do before simplifying.
+
+    Returns
+    -------
+    signomials : list of simplified Signomials
+        Signomials with cs that are solely nans and/or zeroes are removed.
+
+    mmaps : Map from initial monomials to substitued and simplified one.
+            See small_scripts.sort_and_simplify for more details.
+    """
+    signomials_, smaps = [], []
+    for s in signomials:
+        _, exps, cs, _ = substitution(s, subs)
+        # remove any cs that are just nans and/or 0s
+        notnan = ~np.isnan(cs)
+        if np.any(notnan) and np.any(cs[notnan] != 0):
+            exps, cs, smap = simplify_exps_and_cs(exps, cs, return_map=True)
+            if s.any_nonpositive_cs:
+                negative_c_count = (cs <= 0).sum()
+                if negative_c_count == 0:
+                    raise RuntimeWarning("""Infeasible SignomialConstraint.
+
+    %s became infeasible  when all negative terms were substituted out.""" % s)
+                elif negative_c_count == 1:
+                    # turn it into a Posynomial constraint
```

Comment:
I'm confused here because our implementation seems to assume that `signomials` is really a list of constraints. Is this function also intended to work with raw `Signomial` objects (non-constraints), as the docstring suggests? If so, I'm confused about how those are handled.